tests/components/kodi/test_config_flow.py

```diff
@@ -0,0 +1,325 @@
+"""Test the Kodi config flow."""
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.kodi.config_flow import (
+    CannotConnectError,
+    InvalidAuthError,
+)
+from homeassistant.components.kodi.const import DEFAULT_TIMEOUT, DOMAIN
+
+from .util import (
+    TEST_CREDENTIALS,
+    TEST_DISCOVERY,
+    TEST_HOST,
+    TEST_IMPORT,
+    TEST_WS_PORT,
+    UUID,
+    MockConnection,
+)
+
+from tests.async_mock import AsyncMock, patch
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+async def user_flow(hass):
+    """Return a user-initiated flow after filling in host info."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], TEST_HOST
+    )
+
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    return result["flow_id"]
+
+
+@pytest.fixture
+async def discovery_flow(hass):
+    """Return a discovery flow after confirmation."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "zeroconf"}, data=TEST_DISCOVERY
+    )
+    assert result["type"] == "form"
+    result = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+    return result["flow_id"]
+
+
+async def test_user_flow(hass, user_flow):
+    """Test a successful user initiated flow."""
+    with patch(
+        "homeassistant.components.kodi.config_flow.Kodi.ping", return_value=True,
+    ), patch(
+        "homeassistant.components.kodi.config_flow.get_kodi_connection",
+        return_value=MockConnection(),
+    ), patch(
+        "homeassistant.components.kodi.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.kodi.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_configure(
+            user_flow, TEST_CREDENTIALS
+        )
+
+        assert result["type"] == "form"
+        assert result["errors"] == {}
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], TEST_WS_PORT
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == TEST_HOST["host"]
+    assert result2["data"] == {
+        **TEST_HOST,
+        **TEST_CREDENTIALS,
+        **TEST_WS_PORT,
+        "name": None,
+        "timeout": DEFAULT_TIMEOUT,
+    }
+
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)